### PR TITLE
ci/docs(releasing): updates for release candidates

### PR DESCRIPTION
**Pull request by vdice**
CI and docs updates around release candidates.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -264,7 +276,9 @@ jobs:
     name: create tag sdk/go/v*
     runs-on: ubuntu-latest
     needs: build-and-sign
-    if: startsWith(github.ref, 'refs/tags/v')
+    if: |
+      startsWith(github.ref, 'refs/tags/v') &&
+      needs.build-and-sign.outputs.prerelease == 'false'
```
Comment (vdice) [inline .github/workflows/release.yml:281]: @itowlson I've opted to skip: pusthing the `sdk/go/v*` tag, creating the template SDK updates PR and pushing the templates tag if the v* release is a pre-release/release candidate.  WDYT?

Comment (itowlson) [reply .github/workflows/release.yml:281]: That will make them use main, which is _potentially_ problematic if the main templates/SDKs diverge from the stabilising templates/SDKs.  But hopefully any such changes would be planned for backport anyway.

Comment (vdice) [reply .github/workflows/release.yml:281]: Good point.  We _could_ do all of the above as usual for release candidates but I am unsure about the last one: pushing the templates tag.  We'd push eg `spin/templates/v2.0` on our first v2.0.0-rc.1 release as it stands now.  What might be the best strategy for this bit?

Comment (vdice) [reply .github/workflows/release.yml]: Updated to push sdk + templates tags as usual in https://github.com/fermyon/spin/pull/1942/commits/36e78370b759357b545f97fce792a60fc9bad4aa

**Review by michelleN — APPROVED** [review]

**Review by fibonacci1729 — APPROVED** [review]
Looks great, thanks @vdice !

**Pull request merged**